### PR TITLE
fix(PP-49): play session end

### PR DESCRIPTION
**Pull request by DmitrijSidorkin**

### CODE REVIEW

File: controllers/main.js [hunk 1]
```diff
@@ -105,6 +108,7 @@ module.exports.sendPlayData = async (req, res, next) => {
     difficulty: req.body.difficulty,
     length: req.body.sessionLength,
     sessionData: [],
+    sessionEnd: false,
```
Comment (elnino97) [inline controllers/main.js:111]: Shouldn't this be `sessionEnded`?

File: controllers/main.js [hunk 2]
```diff
@@ -113,29 +117,38 @@ module.exports.sendPlayData = async (req, res, next) => {
...
+      {
+        $set: {
+          "sessionData.$[elem].userGuess": userGuess,
+          "sessionData.$[elem].userGuessText": req.body.guess,
+        },
+      },
+      { arrayFilters: [{ "elem._id": elemId }] }
+    );
+    const playSessionData = await fetchPlaySessionData(req.user.username);
+    if (req.body.action === "back") {
```
Comment (elnino97) [inline controllers/main.js:140]: Would be nice to have these action types declared separately somwhere.

```
const PLAY_ACTION_TYPES = {
  back: 'back'
}
```

File: controllers/main.js (hunk 1 above)
Comment (DmitrijSidorkin) [reply controllers/main.js:111]: Yep, made a typo here, thanks for pointing it out

File: controllers/main.js [hunk 3]
```diff
@@ -113,29 +117,38 @@ module.exports.sendPlayData = async (req, res, next) => {
...
+  if (thisSession.sessionEnded) {
     res.redirect("/results");
+  } else {
+    const pageNum = parseInt(req.body.pageNum);
+    const playSession = await PlaySession.findOne({ sessionId: sessionId });
+    const userGuess =
+      req.body.guess === playSession.sessionData[pageNum - 1].gameName;
+    const elemId = req.body.elemId;
+    await PlaySession.updateOne(
+      { sessionId: sessionId },
```
[the remark this answers is not in the document]
Comment (DmitrijSidorkin) [reply controllers/main.js:130]: I always end up beating around the bush when it comes to mongoDB cause am a bit unsure whether Javascript syntax always work the way you'd expect with it. Guess should get a bit more experimental here and try to figure out what works and what doesn't and potentially save myself some time and make the code more readable.

File: controllers/main.js (hunk 2 above)
Comment (DmitrijSidorkin) [reply controllers/main.js:140]: Could you elaborate a bit more on this one?

**Review by elnino97 — APPROVED** [review]

**Pull request merged**